### PR TITLE
fix(dev): CTL-1628 ADR-026 precision + daemon comment-wake transition-event dispositions (Codex #2961 post-merge)

### PR DESCRIPTION
**Pull request by ryanrozich**
## Summary

Post-merge follow-up to #2961. Started as a docs-precision fix and grew a code fix: the docs verification pass exposed a real bug in the daemon's `worker.transition` emission.

**Code fix — malformed transition event (Codex #2970 round 2):**
- `handleCommentWake`'s needs-human clear (`execution-core/daemon.mjs:512-517`) called `appendWorkerTransitionEvent({ ticket, from: "needs-human", to: "cleared", reason })` — but `buildWorkerTransitionEvent` accepts `fromDisposition`/`toDisposition`, not `from`/`to`. The `from`/`to` keys were silently ignored, so every needs-human-cleared event shipped with both disposition fields `null`. Fixed the call site to pass `fromDisposition: "needs-human"`, `toDisposition: null` (the builder auto-encodes this as the `"cleared"` sentinel), and added `orchId` for schema parity with the sibling needs-input clear site. Added a regression test in `daemon.test.mjs` asserting the emitted envelope carries the correct dispositions and that the old `{from, to}` keys don't resurface.
- Retitled this PR `fix(dev)` so the bugfix ships with a catalyst-dev patch version bump.

**Docs precision, round 1 (Codex #2961 post-merge):**
- **Chokepoint scope**: scoped ADR-026's "all worker state transitions" claim to scheduler-owned transitions, naming the daemon's `handleCommentWake` as a known, self-documented second emitter (mirrored into `docs/architecture.md`).
- **Sink-5 overclaim**: verified `ticket_state_transitions` has zero schema/writer/consumer anywhere in the codebase — corrected both docs to say sink 5 (the broker table) was never implemented; only sink 4 (OTLP) is live.

**Docs precision, round 2 (Codex #2970):**
- `references/event-schema.md`'s `worker.transition` fields section claimed a single producer (`recordTransition`); added the daemon's comment-wake path as a second, real producer.
- ADR-026's Consequences section still advertised "five sinks... optional broker table" — reconciled with the sink-5-never-shipped correction (now: four live sinks, sink 5 never implemented).
- Verified `killHungWorker` (Pass 0w's hung-worker escalation, `watchdog-action.mjs`) applies `needs-human` via `labelNeedsHumanUnlessBeliefOwner` with **no** `worker.transition` emission anywhere in that path — a genuine coverage gap, distinct from the daemon's deliberate second-producer pattern. Documented it as such in both `docs/adrs.md` and `docs/architecture.md`, and reworded the ADR-026 "Decision" claim to "transitions the scheduler records" for precision.

## Test plan

- [x] `ctl-764-docs.test.sh` — 14/14 pass
- [x] `daemon.test.mjs` — 148/148 pass (incl. new regression test for the fixed call site)
- [x] `worker-transition-event.test.mjs` — pass
- [x] `watchdog-action.test.mjs` — 13/13 pass
- [x] `integration-ctl-768.test.mjs` — 1 pre-existing unrelated failure, confirmed via `git stash` against the branch tip before this change
- [x] Repo-wide grep confirms zero schema/writer/consumer for `ticket_state_transitions`
- [x] `node --check` on `daemon.mjs`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -429,25 +429,36 @@ at explicit resolution (Done or terminal-sweep-clear), not on steady-state ticks
...
+writer, no broker consumer exist anywhere in the codebase; it remains a planned item, not a shipped
+one. The remaining scheduler-side sinks are written at their own scheduler sites around the same
+transition (not fanned out from inside the chokepoint): (1) Linear Status via the
+`applyPhaseStatus` chokepoint (Axis 1), (2) the `worker-status` label via the admission converger
+(`convergeHeldLabel`) / `labelOnce` (Axis 2).
+
+**The scheduler chokepoint is not the only `worker.transition` emitter.** The daemon's
+`handleCommentWake` (CTL-768, `execution-core/daemon.mjs`) calls `appendWorkerTransitionEvent`
+directly at two sites — clearing a stale `needs-human` marker on human reply, and clearing
+`needs-input` after a comment-driven wake — bypassing `recordTransition` entirely. Its own code
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply docs/architecture.md]: Confirmed — this was a real bug your docs work exposed. `handleCommentWake`'s needs-human clear (`daemon.mjs:512-517`) passed `{from, to}` to `appendWorkerTransitionEvent`, but `buildWorkerTransitionEvent` only accepts `fromDisposition`/`toDisposition` — the extra keys were silently dropped, so every emitted event carried both disposition fields as `null`. Checked the sibling call site at :624-632 too — that one already used `fromDisposition`/`toDisposition` correctly, so it was only the needs-human clear that was broken. Fixed the call site to pass `fromDisposition: "needs-human"`, `toDisposition: null` (auto-encoded as the `"cleared"` sentinel by the builder), added `orchId` for parity with the sibling site, and added a regression test in `daemon.test.mjs` asserting the fixed shape and that the old `{from, to}` keys don'\''t resurface. Own commit 71d91478, PR retitled to `fix(dev)` so it ships with a version bump. Also updated the architecture.md doc sentence to describe the now-well-formed clear.

File: docs/architecture.md [hunk 2]
```diff
@@ -429,25 +429,36 @@ at explicit resolution (Done or terminal-sweep-clear), not on steady-state ticks
...
+`ticket_state_transitions` table (CTL-764 Phase 10), was **never implemented** — no schema, no
+writer, no broker consumer exist anywhere in the codebase; it remains a planned item, not a shipped
+one. The remaining scheduler-side sinks are written at their own scheduler sites around the same
+transition (not fanned out from inside the chokepoint): (1) Linear Status via the
+`applyPhaseStatus` chokepoint (Axis 1), (2) the `worker-status` label via the admission converger
+(`convergeHeldLabel`) / `labelOnce` (Axis 2).
+
+**The scheduler chokepoint is not the only `worker.transition` emitter.** The daemon's
+`handleCommentWake` (CTL-768, `execution-core/daemon.mjs`) calls `appendWorkerTransitionEvent`
+directly at two sites — clearing a stale `needs-human` marker on human reply, and clearing
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply docs/architecture.md]: Fixed in commit 5035b67a — added the daemon comment-wake path as a second producer to the `worker.transition` fields section in references/event-schema.md, consistent with the architecture.md wording from the prior round.

File: docs/adrs.md [hunk 3]
```diff
@@ -514,19 +514,29 @@ resolvable stalls consuming attention); leave re-engagement to the inference eng
...
+`recordWorkerTransition` module (`record-worker-transition.mjs`) named in this ADR's original
+design: that module's own doc comment declared only three of the eventual five sinks (Sink 1 Linear
+workflow status via `applyPhaseStatus`, Sink 2 the disposition label via `convergeLabel`, Sink 3 the
+unified event log via `appendWorkerTransitionEvent`) and flagged itself as unfinished — "Phase 5 will
+wire the production defaults... and route all call sites here." That Phase 5 wiring never happened;
+the scheduler's live path never called the module. Of the two sinks the module never reached, only
+sink 4 (OTLP via `otel-forward`) is actually live — it rides automatically on sink 3's event-log
+write. Sink 5 (a broker `ticket_state_transitions` table, CTL-764 Phase 10) was **never
+implemented** anywhere, live path or otherwise — no schema, no writer, no broker consumer exist in
+the codebase. CTL-1628 removed the retired module as consumer-free.
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply docs/adrs.md:531]: Fixed in commit 5035b67a — reconciled ADR-026's Consequences section with the sink-5-never-shipped correction: it now says four **live** sinks (Linear Status, label, event log, OTLP) and explicitly states the fifth (the broker ticket_state_transitions table, CTL-764 Phase 10) was designed but never implemented — no schema, writer, or consumer exist anywhere in the codebase.

File: docs/adrs.md [hunk 4]
```diff
@@ -514,19 +514,29 @@ resolvable stalls consuming attention); leave re-engagement to the inference eng
...
-`appendWorkerTransitionEvent`) and flagged itself as unfinished — "Phase 5 will wire the production
-defaults... and route all call sites here." That Phase 5 wiring never happened; the scheduler's live
-path never called the module, and OTLP forwarding (sink 4) and the broker `ticket_state_transitions`
-table (sink 5) were added directly to the live inline path, never retrofitted into it. CTL-1628
-removed it as consumer-free. See "Two-axis worker state & the recordWorkerTransition chokepoint
-(CTL-764)" in `docs/architecture.md` for the live mechanism.
+**Decision** — all **scheduler-owned** worker state transitions are consolidated behind a single
+chokepoint and a workspace-scoped, single-valued `worker-status` Linear label group carrying worker
+_disposition_ independently of _pipeline stage_. The live chokepoint is the **inline
+`recordTransition`** function inside `scheduler.mjs`'s `schedulerTick` — not the standalone
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply docs/adrs.md]: Verified — confirmed via killHungWorker (watchdog-action.mjs), its scheduler.mjs call site (Pass 0w), and label-guard.mjs: labelNeedsHumanUnlessBeliefOwner applies the needs-human label but neither it nor killHungWorker nor the scheduler.mjs Pass-0w call site invoke recordTransition, appendWorkerTransitionEvent, or any other event emitter. Unlike the daemon's handleCommentWake (a deliberate second producer), this is a genuine coverage gap — a real Axis-2 transition with no worker.transition record at all. Fixed in commit 5035b67a: reworded ADR-026's Decision claim to "transitions the scheduler records" for precision, and documented this as a named, distinct gap (not a second producer) in both docs/adrs.md and docs/architecture.md, alongside the daemon producer paragraph.

File: plugins/dev/scripts/execution-core/daemon.mjs [hunk 5]
```diff
@@ -511,8 +511,9 @@ export async function handleCommentWake(
     try {
       appendWorkerTransitionEvent({
         ticket,
-        from: "needs-human",
-        to: "cleared",
+        orchId: ticket,
+        fromDisposition: "needs-human",
+        toDisposition: null,
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/scripts/execution-core/daemon.mjs]: Confirmed and fixed in commit 26a616cb. Verified in linear-write.mjs:removeLabel — the idempotent no-op path (label already absent) and the real-write path both returned {removed:true}, with no way to tell them apart. Extended the return shape additively with a wrote field (wrote:false on the no-op and on all failure paths, wrote:true only when the removal actually executed and succeeded), keeping removed's existing semantics untouched for every other caller. Gated the needs-human clear's transition emission on clearedNeedsHumanWrote (a new variable separate from clearedNeedsHuman, which still gates the marker-reconcile block — that must run on a no-op too, to keep labelOnce in step). Also applied the same wrote-based gate to the needs-input clear site for consistency (it had the identical ambiguity, just with an existing 'finding E' comment defending the old removed-only semantics as testable-but-imprecise) — flagging that as an extra-scope decision beyond what you asked for the needs-human site specifically. Added regression tests for both the write and no-op paths at both call sites (daemon.test.mjs) plus wrote assertions on linear-write.test.mjs's existing no-op/real-write cases.

[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/scripts/execution-core/daemon.mjs]: Verified — daemon.mjs imports startScheduler directly from ./scheduler.mjs, so the daemon and scheduler run in the SAME process, sharing the SAME live lastDispositionEmit Map instance (Node's ESM module cache). Confirmed the risk is real: needs-input already self-heals via an unconditional check on the scheduler's next relevant tick (the `lastDispositionEmit.get(ticket) === HELD_LABEL_NEEDS_INPUT` branch), but needs-human has no equivalent — its three scheduler-side self-heal paths (terminal-done-clear, terminal-sweep-clear, no-stall-clear, all via clearStalledLabel's onRemoved -> recordTransition) are gated on the SAME once-marker files that the daemon's clearNeedsHumanMarkers already deletes, so they never even attempt the clear until (if ever) the ticket reaches terminal Done. Fixed in commit 26a616cb: exported clearDispositionEmit(ticket) from scheduler.mjs (a thin lastDispositionEmit.set(ticket, null)), made it injectable in handleCommentWake (matching the existing appendWorkerTransitionEvent seam pattern), and called it from both the needs-human site (gated on clearedNeedsHumanWrote — only a real write) and the needs-input site (belt-and-suspenders alongside its existing self-heal, closing the window immediately instead of waiting a tick). Added regression tests asserting the reset fires on a real write and does NOT fire on a no-op.

File: docs/adrs.md [hunk 6]
```diff
@@ -559,10 +577,12 @@ merged axes into a single status enum (rejected: pipeline stage and disposition
 both need independent observability); async `recordWorkerTransition` only (rejected: `schedulerTick`
 is sync; async would require a separate flush loop with new failure modes).
 
-**Consequences** — every transition fans out to five sinks (Linear Status, label, event log, OTLP
-via otel-forward, optional broker table); all fail-open. The HUD capacity header gains
-per-disposition buckets and triage is carved out of `maxParallel` counting. AGENTS.md /
-architecture.md carry the two-axis model as first-class concepts.
+**Consequences** — every scheduler-owned transition fans out to four **live** sinks (Linear Status,
+label, event log, OTLP via otel-forward); all fail-open. A fifth sink (an optional broker
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply docs/adrs.md]: Confirmed and fixed in commit 0e376956. Read the dependency-cycle path (scheduler.mjs ~5666-5678): it calls labelNeedsHumanUnlessBeliefOwner (label sink) and recordTransition({ toDisposition: "needs-human" }) with no toStage/fromStage at all — so it reaches label + event log + OTLP (3 sinks) but never applyPhaseStatus (Linear Status), since there's no stage change to record. More generally: recordTransition's own contract means no single call reaches all four sinks unless it sets BOTH toStage and toDisposition together — a pure stage move skips the label sink, a pure disposition move skips Linear Status. Rewrote the Consequences section to say four sinks are live but no single transition reaches all four, explained the stage-only vs disposition-only split, and cited the dependency-cycle path as the concrete disposition-only example.

File: docs/architecture.md [hunk 7]
```diff
@@ -429,25 +429,44 @@ at explicit resolution (Done or terminal-sweep-clear), not on steady-state ticks
...
+`needs-input` after a comment-driven wake — bypassing `recordTransition` entirely. Its own code
+comment says why: "scheduler.mjs owns the park/apply emission; the clear is emitted here (the daemon
+removes the durable label out-of-band and redispatches — the scheduler never observes this edge)."
+This is a deliberate, self-documented second producer, not a gap in the chokepoint design.
+
+**A separate escalation path emits no `worker.transition` at all.** Pass 0w's hung-worker
+escalation (`killHungWorker` in `watchdog-action.mjs`, invoked from `scheduler.mjs`'s
+progress-watchdog pass) applies the `needs-human` label via `labelNeedsHumanUnlessBeliefOwner`
+(`label-guard.mjs`) but never calls `recordTransition`, `appendWorkerTransitionEvent`, or any other
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply docs/architecture.md]: Confirmed and fixed in commit 0e376956. killHungWorker (watchdog-action.mjs:144-152, default emitter at :53) does emit phase.terminal.reap-requested via emitReapIntent when bgJobId exists — that part of the kill/reap sequence is observable. Rescoped both docs' claim from "emits no worker.transition at all" (which read as "emits nothing") to "emits no worker.transition for its disposition change" — explicitly noting the reap-intent emission exists and is a separate, unrelated event, while the specific gap is the missing recordTransition/appendWorkerTransitionEvent call for the needs-human escalation itself.\n\nOn CTL-1638: checked its description — it already scopes the claim narrowly ("ZERO worker.transition emission... no recordTransition, no appendWorkerTransitionEvent"), not the broader "nothing observable" claim the docs had. It doesn't need the same correction; no ticket update needed on this specific point.

File: docs/architecture.md [hunk 8]
```diff
@@ -429,25 +429,44 @@ at explicit resolution (Done or terminal-sweep-clear), not on steady-state ticks
...
+`applyPhaseStatus` chokepoint (Axis 1), (2) the `worker-status` label via the admission converger
+(`convergeHeldLabel`) / `labelOnce` (Axis 2).
+
+**The scheduler chokepoint is not the only `worker.transition` emitter.** The daemon's
+`handleCommentWake` (CTL-768, `execution-core/daemon.mjs`) calls `appendWorkerTransitionEvent`
+directly at two sites — clearing a stale `needs-human` marker on human reply, and clearing
+`needs-input` after a comment-driven wake — bypassing `recordTransition` entirely. Its own code
+comment says why: "scheduler.mjs owns the park/apply emission; the clear is emitted here (the daemon
+removes the durable label out-of-band and redispatches — the scheduler never observes this edge)."
+This is a deliberate, self-documented second producer, not a gap in the chokepoint design.
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply docs/architecture.md]: Confirmed and fixed in commit 0e376956. Traced the two sites' control flow: the needs-input clear is a per-signal branch gated on status === "needs-input" and calls dispatch(orchDir, ticket, parkedPhase, {...}) right after the transition emission, in the same block — the quoted "redispatches" rationale fits it exactly. The needs-human clear runs once per comment-wake call, gated on humanProvenance && isManagedTicket, with no dispatch call anywhere in that block. Split the single shared quote into two bullets in both docs: the needs-input clear keeps the original quote verbatim (it's accurate there), and the needs-human clear gets its own rationale (the scheduler's own STICKY needs-human handling explicitly defers clearing to a confirmed-removal signal rather than clearing it itself on a steady-state pass) with an explicit note that the "redispatches" half doesn't apply to that site.

**Comment by ryanrozich** [comment]
@codex review

File: docs/architecture.md [hunk 9]
```diff
@@ -429,25 +429,44 @@ at explicit resolution (Done or terminal-sweep-clear), not on steady-state ticks
...
+`ticket_state_transitions` table (CTL-764 Phase 10), was **never implemented** — no schema, no
+writer, no broker consumer exist anywhere in the codebase; it remains a planned item, not a shipped
+one. The remaining scheduler-side sinks are written at their own scheduler sites around the same
+transition (not fanned out from inside the chokepoint): (1) Linear Status via the
+`applyPhaseStatus` chokepoint (Axis 1), (2) the `worker-status` label via the admission converger
+(`convergeHeldLabel`) / `labelOnce` (Axis 2).
+
+**The scheduler chokepoint is not the only `worker.transition` emitter.** The daemon's
+`handleCommentWake` (CTL-768, `execution-core/daemon.mjs`) calls `appendWorkerTransitionEvent`
+directly at two sites — clearing a stale `needs-human` marker on human reply, and clearing
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply docs/architecture.md]: Confirmed and fixed in commit 73ab50f3. The old table cell and precedence text said needs-human is cleared ONLY by clearStalledLabel, contradicting the newer producer-split paragraph. Reconciled both docs: needs-human now has TWO documented removal paths — (1) clearStalledLabel's onRemoved at scheduler-side resolution points (Done/terminal-sweep/no-stall-clear), and (2) the daemon's handleCommentWake needs-human clear on a managed ticket's confirmed human reply (write-gated via the wrote field, emission-carrying via appendWorkerTransitionEvent, and resets the scheduler's lastDispositionEmit dedup via clearDispositionEmit — all from this PR's round 3/4 fixes). Updated the Cleared-by table cell, the Precedence sentence, and the Resolution-gated-clearing paragraph in both docs/architecture.md and docs/adrs.md consistently.

File: plugins/dev/references/event-schema.md [hunk 10]
```diff
@@ -661,7 +661,12 @@ emitting it during a mixed-version fleet rollout.
 
 #### worker.transition fields (CTL-764)
 
-Emitted by `recordTransition` in `scheduler.mjs` (sync chokepoint). Dims are **attributes** (not
+Emitted by two producers: `recordTransition` in `scheduler.mjs` (the sync chokepoint for
+scheduler-owned transitions), and the daemon's `handleCommentWake` (CTL-768,
+`execution-core/daemon.mjs`), which calls `appendWorkerTransitionEvent` directly at two sites — the
+`needs-human`→cleared resolution on a human reply, and the `needs-input`→cleared resolution after a
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/references/event-schema.md]: Confirmed, and this is a REAL, unrecorded emission gap — same class as Pass 0w, not fixed in this PR (see below). Traced the exact mechanism: when a comment wake finds no local worker dir (readdirSync(workerDir) throws), the function catches and returns immediately with the comment "No local worker state: the label clear above is the entire correct response." That early return happens AFTER the needs-human block's SECONDARY, uncaptured removeLabel(ticket, "needs-input") cleanup call (the "both labels are the same fact" cleanup) but BEFORE the per-signal loop that contains the only needs-input worker.transition emission. So on a ticket with no local worker dir but a stale durable needs-input Linear label, the daemon DOES clear that label but NEVER emits a worker.transition for it — the per-signal-loop emission path is signal-backed only. The existing 'REGRESSION: clears needs-human even when the ticket has NO worker directory' test's own comment cites this as a common real-world case (10 of 12 parked tickets on the live fleet 2026-07-29 had no worker dir on either host), so this isn't a rare edge case.\n\nScoped the event-schema.md producer claim to say the needs-input emission is signal-backed-only, spelled out the no-worker-dir gap explicitly, and flagged it as Pass-0w-class. Please extend CTL-1638 (or file a sibling ticket) to cover this — fixing the actual behavior (either emitting from the needs-human block's secondary removeLabel call, or restructuring so the early return doesn't skip it) is more than a docs-only round should take on.

**Comment by ryanrozich** [comment]
@codex review

File: plugins/dev/scripts/execution-core/daemon.mjs [hunk 11]
```diff
@@ -605,13 +637,18 @@ export async function handleCommentWake(
...
+    // idempotent already-absent case (linear-write.mjs) — gate the emission on the
+    // additive `wrote` field instead, so a duplicate webhook / second host re-checking an
+    // already-cleared label doesn't record a second "cleared" transition for a change
+    // that happened on a prior call. `wrote` is undefined on a test-stub `undefined`
+    // return, so the `res === undefined` success path stays gated on that alone.
     let needsInputRemoved = false;
     try {
       const res = await removeLabel(ticket, "needs-input"); // CTL-764 Phase 4: durable needs-input cleared on genuine resolution
-      needsInputRemoved = res === undefined || res?.removed === true;
+      needsInputRemoved = res === undefined || res?.wrote === true;
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/scripts/execution-core/daemon.mjs:651]: Fixed in #2973 (commit 69105591). Read the full flow across all three interacting sites: the earlier needs-human-block cleanup (~:488-497), the no-worker-dir early return (the CTL-1638-extended gap), and the per-signal loop's own removeLabel call. Confirmed the race exactly as you described: the earlier needs-input removeLabel call's result was discarded, so when IT performed the real write, the later per-signal-loop call observed {wrote:false} and skipped the emission the earlier write earned. Captured the earlier call's wrote result into needsInputWroteEarly and OR'd it into the per-signal loop's emission gate — a genuine write anywhere in the invocation now produces exactly one emission. Added a regression test that simulates the exact race (first removeLabel(ticket,"needs-input") call returns wrote:true, second returns wrote:false) and asserts exactly one emission fires, plus a companion test for the fully-cross-host case (neither call in this invocation wrote) asserting zero emissions.

File: plugins/dev/scripts/execution-core/daemon.mjs [hunk 12]
```diff
@@ -508,15 +522,33 @@ export async function handleCommentWake(
...
-        to: "cleared",
-        reason: "human-responded",
-      });
-    } catch {
-      /* observability only */
+    // Codex #2970 round 3: gate on a CONFIRMED write, not merely a confirmed-absent
+    // read — otherwise a duplicate webhook / second host that finds the label
+    // already gone would still record a fresh "cleared" transition for a change
+    // that happened on a prior call, mislabeling the timeline.
+    if (clearedNeedsHumanWrote) {
```
[the remark this answers is not in the document]
Comment (ryanrozich) [reply plugins/dev/scripts/execution-core/daemon.mjs:529]: Fixed in #2973 (commit 69105591). Split the write-gated clearedNeedsHumanWrote (still gates the transition EMISSION only — only the writer host emits, so no duplicate 'cleared' records) from clearedNeedsHuman (confirmed removal regardless of writer, now gates clearDispositionEmit directly). A cross-host {removed:true, wrote:false} observation now correctly resets this process's stale lastDispositionEmit entry even though this process didn't do the write. Applied the identical split to the needs-input site for consistency — introduced needsInputConfirmed (removed !== false, from the per-signal loop's own removeLabel result) separate from needsInputRemoved (write-only, now also folding in the round-1 fix's needsInputWroteEarly). Added regression tests for both needs-human (cross-host confirmed clear resets dedup with no emission; an unconfirmed removed:false does NOT reset) and needs-input (fully cross-host no-op resets dedup with no emission).
